src/components/GroupModal/LoansGroupModal.tsx

```diff
@@ -54,14 +64,43 @@ const LoansGroupModal: FC<LoansGroupModalProps> = ({
     );
   }, [close, materialsToRenew, modalIdUsed, mutate, queryClient]);
 
+  const renewSelected = useCallback(() => {
+    const selectedLoansDueDate = loansModal
+      .filter(({ loanId }) => materialsToRenew.includes(String(loanId) || ""))
+      .map(({ dueDate: localDueDate }) => localDueDate)
+      .filter((item) => item !== undefined && item !== null);
+    const acceptModal =
+      loansOverdue(loansModal) && sameDueDate(selectedLoansDueDate as string[]);
+
+    if (acceptModal) {
+      openAcceptModal();
+    } else if (!acceptModal) {
+      renew();
+    }
+  }, [loansModal, materialsToRenew, openAcceptModal, renew]);
+
   useEffect(() => {
     setMaterialsToRenew(getRenewableMaterials(loansModal));
   }, [loansModal]);
 
+  useEffect(() => {
+    if (accepted) {
+      setAcceptedButtonPressed(accepted);
+      resetAccepted();
+    }
+  }, [accepted, resetAccepted]);
+
   const selectMaterials = (materialIds: string[]) => {
     setMaterialsToRenew(materialIds);
   };
 
+  useEffect(() => {
+    if (acceptedButtonPressed) {
+      renew();
+      setAcceptedButtonPressed(false);
+    }
+  }, [acceptedButtonPressed, renew]);
+
```

Comment:
This seems funky to me. Could be that it makes sense. But having ìf acceptedButtonPressed then renew() and  setAcceptedButtonPressed(false)` instead of sorting that out in a clickhandler has a code smell to me. Could be wrong though :)

Reply:
it was to keep the renew function in _one_ file.